File: ooo.kt
val LINE = (1..50).map { '-' }.joinToString(separator = "")

fun main(args: Array<String>) {
    I.willBe().ooo().`for`(5).minutes()
}

class I private constructor(private var minutes: Int? = null) {
    init { println(LINE) }
    companion object { fun willBe() = I() }

    fun ooo() = builderBlock { print("ooo ") }

    fun `for`(minutes: Int) = builderBlock { this.minutes = minutes }

    fun minutes() {
        println("$minutes min")
        println(LINE)
    }

    private fun builderBlock(f: () -> Unit): I {
        f()
        return this
    }
}

File: ooo.dart
import 'dart:io';
import 'dart:async';
import 'dart:math';

void main() async {
  await I()..willBe()..ooo()..for_(minutes: 5);
}

class I {
  void willBe() { /* pass */ }

  void ooo() {
    stdout.write("ooo ");
  }

  Future<void> for_({int minutes}) async {
    Future.any([
      Future.value(_someMethod(minutes)),
      Future.value(_someMethod(minutes)),
      Future.value(_someMethod(minutes))
    ]).asStream().listen((min) {
      print('$min min');
    });
  }

  Future<int> _someMethod(int value) async {
    final Stream<int> stream = Stream.fromIterable(Iterable.generate(50000, (i) => Random().nextInt(value + 1)));
    await for (var i in stream) {
      if(value == i)
        return i;
    }
    return value;
  }
}